tests/lib/ocpp/v201/test_composite_schedule.cpp

```diff
@@ -0,0 +1,431 @@
+
+        return device_model;
+    }
+
+    void create_evse_with_id(int id) {
+        testing::MockFunction<void(const MeterValue& meter_value, const Transaction& transaction, const int32_t seq_no,
+                                   const std::optional<int32_t> reservation_id)>
+            transaction_meter_value_req_mock;
+        testing::MockFunction<void()> pause_charging_callback_mock;
+        auto e1 = std::make_unique<Evse>(
+            id, 1, *device_model, database_handler, std::make_shared<ComponentStateManagerMock>(),
+            transaction_meter_value_req_mock.AsStdFunction(), pause_charging_callback_mock.AsStdFunction());
+        evses[id] = std::move(e1);
+    }
+
+    SmartChargingHandler create_smart_charging_handler() {
+        return SmartChargingHandler(evses, device_model);
+    }
+
+    std::string uuid() {
+        std::stringstream s;
+        s << uuid_generator();
+        return s.str();
+    }
+
+    void open_evse_transaction(int evse_id, std::string transaction_id) {
+        auto connector_id = 1;
+        auto meter_start = MeterValue();
+        auto id_token = IdToken();
+        auto date_time = ocpp::DateTime("2024-01-17T17:00:00");
+        evses[evse_id]->open_transaction(
+            transaction_id, connector_id, date_time, meter_start, id_token, {}, {},
+            std::chrono::seconds(static_cast<int64_t>(1)), std::chrono::seconds(static_cast<int64_t>(1)),
+            std::chrono::seconds(static_cast<int64_t>(1)), std::chrono::seconds(static_cast<int64_t>(1)));
+    }
+
+    void install_profile_on_evse(int evse_id, int profile_id) {
+        if (evse_id != STATION_WIDE_ID) {
+            create_evse_with_id(evse_id);
+        }
+        auto existing_profile = create_charging_profile(profile_id, ChargingProfilePurposeEnum::TxDefaultProfile,
+                                                        create_charge_schedule(ChargingRateUnitEnum::A), uuid());
+        handler.add_profile(evse_id, existing_profile);
+    }
+
+    void log_me(std::vector<ChargingProfile> profiles) {
+        EVLOG_info << "[";
+        for (auto& profile : profiles) {
+            EVLOG_info << "  ChargingProfile> " << utils::to_string(profile);
+        }
+        EVLOG_info << "]";
+    }
+
+    // Default values used within the tests
+    std::map<int32_t, std::unique_ptr<EvseInterface>> evses;
+    std::shared_ptr<DatabaseHandler> database_handler;
+
+    sqlite3* db_handle;
+
+    bool ignore_no_transaction = true;
+    std::shared_ptr<DeviceModel> device_model = create_device_model();
+    SmartChargingHandler handler = create_smart_charging_handler();
+    boost::uuids::random_generator uuid_generator = boost::uuids::random_generator();
+};
+
+/**
+ * Validates the SmartChargingHandler::determined_duraction and SmartChargingHandler::within_time_window
+ * utility functions.
+ */
+TEST_F(ChargepointTestFixtureV201, K08_CalculateCompositeSchedule_DetermineDurationAndWithinTimeWindow) {
+    GTEST_SKIP();
```

Comment:
Should these tests have `GTEST_SKIP()`?

Reply:
Removed all but one that will need to triage.